opentelemetry-otlp/src/span.rs

```diff
@@ -336,6 +425,46 @@ impl SpanExporter for TraceExporter {
 
                 Ok(())
             }
+
+            #[cfg(feature = "http-proto")]
+            TraceExporter::HTTP {
+                trace_exporter,
+                collector_endpoint,
+                headers,
+                ..
+            } => {
+                let req = TonicRequest {
+                    resource_spans: batch.into_iter().map(Into::into).collect(),
+                };
```

Comment:
Does this mean the `http-proto` will depends on the `tonic`? We should probably figure out how to generate proto file with `prost` only and use it for `tonic` and `http-proto`

Reply:
TonicRequest is just an alias to the ExportTraceServiceRequest (at the top of teh file), should we use another name ?
I believe in order to generate the proto rs files, the tonic-build is used, that is listed under build-dependencies and it is not available in the package itself.

Reply:
Yeah I think it would be reasonable to list the `tonic_build` as a dependency of `http-proto` as it was marked optional.

Reply:
I see your point and I don't have a strong opinion on this. The tonic-build is a slim package that includes prost-build, for consistency I would use it, but we could try to build the protos with prost-build if you feel that it is better.

Reply:
also should I remove the alias name TonicRequest and leave it with the original ExportTraceServiceRequest to eliminate any possible confusions ?

Reply:
You are right, I tweaked it a little bit over the weekend. Looks like the `tonic_build` is a wrapper of `prost-build` so I think we could just use the `tonic_build`. Just need to add it into the dependency.

---
> also should I remove the alias name TonicRequest and leave it with the original ExportTraceServiceRequest to eliminate any possible confusions ?

Yeah that sounds like a good idea 👍  I think those types were named as `TonicXXX` just to distinguish with grpcio generated types.

Reply:
I take my word back, I'm afraid that we should continue to refer to TonicXXX, as tonic remains a dependency as long as the transformations (like SpanData to Tonic ResourceSpans ) and others are constrained by tonic feature utilization. Not sure what would be the best approach here.

Reply:
It would be best if we can avoid `tonic` dependency here as we don't need it for HTTP-based transports. One way to avoid it is to use `prost-build` to build another set of type files. It will requires some changes to the `build.rs`.

Currently, the files generated by `tonic` are located in `OUT_DIR`. So in order to have generated files from `prost-build` and `tonic-build`. We should first try to build them into their respective folders.

And then we should use `include!` to include those generated files in `proto` mod.

Reply:
I agree, we can try it out. The only thing I'm a little afraid is the duplication of code in the "transform" module, as we may have to  create all the transformations (ex from SpanData to whatever resources the prost-build will create), which I expect to look similar to Tonic generated resources.

Reply:
I think it's fine as we already have two transform, one for grpcio and the other one for tonic. But if you feel it's too much for one PR we can merge this one with tonic dependency and use another PR to add transform.

Reply:
I have updated based on your suggestions.